src/twisted/conch/ssh/session.py

```diff
@@ -91,6 +92,33 @@ def request_pty_req(self, data):
         else:
             return 1
 
+    def request_env(self, data):
+        """
+        Process a request to pass an environment variable.
+
+        @param data: The environment variable name and value, each encoded
+            as an SSH protocol string and concatenated.
+        @type data: L{bytes}
+        @return: A true value if the request to pass this environment
+            variable was accepted, otherwise a false value.
+        """
+        if not self.sessionSetEnv:
+            try:
+                self.sessionSetEnv = ISessionSetEnv(self.avatar)
+            except TypeError:
+                log.msg('unhandled request for env')
+                return 0
+        name, value, data = common.getNS(data, 2)
+        try:
+            self.sessionSetEnv.setEnv(name, value)
+        except ValueError:
+            return 0
+        except Exception:
+            log.err()
```

Comment:
As above re: new vs. old style loggers: with a new style logger this would be `_log.failure()`